adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -583,9 +655,22 @@ private boolean isBrokerWithPRTSupport(final Intent intent) {
      */
     public String getCurrentUser() {
         // authenticator is not used if there is not any user
-        Account[] accountList = mAcctManager.getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
-        if (accountList.length > 0) {
-            return accountList[0].name;
+        if (isBrokerAccountServiceSupported()) {
```

Comment:
>isBrokerAccountServiceSupported [](start = 12, length = 31)

Can you make some public method isPermissionsRequieredForBrokerApp that will do the same as isBrokerAccountServiceSupported
So we can use it within app to detect whether we should prompt for permissions